lit_parrot/model.py

```diff
@@ -186,11 +196,15 @@ def forward(
         B, T, C = x.size()  # batch size, sequence length, embedding dimensionality (n_embd)
 
         qkv = self.attn(x)
-        head_size = C // self.n_head
-        qkv = qkv.view(B, T, self.n_head, 3 * head_size).transpose(1, 2)
-        q, k, v = qkv.split(head_size, dim=-1)  # (B, nh, T, hs)
 
-        n_elem = int(self.rotary_percentage * head_size)
+        if self.config.multi_query:
```

Comment:
It would be great to add the gist of what is multi_query in a comment, so readers don't have to decode it from the following lines (ok to do it in the 40B PR)

Reply:
I'm actually removing this in the 40B PR. See description there. But I can add a reference to the original MQA paper